### PR TITLE
chore: upgrade jest and jsdom

### PR DESCRIPTION
**Pull request by jenseng**
This lets us remove a polyfill and some other hacks, and brings along some additional jsdom features which will facilitate testing fixes to submitter serialization (see #9865).

Note:
 * jsdom is making it harder to stub things, so we inject a jest-friendly window into the router. I'm not a huge fan of DI when used solely for the sake of testing, but this seems less terrible than other approaches (e.g. patch-package-ing jsdom, or monkey-patching Object.defineProperty) 🙃
 * We use a yarn resolution to get the latest jsdom, despite the latest jest pinning it at ^20.0.0. Per the maintainers, [this is fine](https://github.com/jestjs/jest/pull/13825#issuecomment-1531649217) as jest doesn't need any code changes to work with new jsdom, and there are no concrete plans yet to ship a new version.

### CODE REVIEW

**Review by MichaelDeBoey — APPROVED** [review]

**Comment by jenseng** [comment]
👋  @brophdawg11 :)

**Comment by brophdawg11** [comment]
Thanks for the ping @jenseng - I kept meaning to come back to this and then, well, didn't 😅 

Let me know if the minor updates look ok to you and I'll get this merged.  Agree on DI being better than the alternatives, and it also follows the pattern we use in `react-router-dom` as well.

**Comment by jenseng** [comment]
Awesome, thanks @brophdawg11!

File: packages/router/__tests__/router-test.ts [hunk 1]
```diff
@@ -429,15 +428,18 @@ function setup({
     });
   }
 
-  // jsdom is making more and more properties non-configurable, so we inject our own jest-friendly window 😅
-  let window = {
-    ...globalWindow,
+  // jsdom is making more and more properties non-configurable, so we inject
+  // our own jest-friendly window.
+  let testWindow = {
```
[the remark this answers is not in the document]
Comment (jenseng) [reply packages/router/__tests__/router-test.ts:433]: LGTM :shipit: 🥳

**Pull request merged**